packages/app/src/debug/DebugPageDetails.vue

```diff
@@ -0,0 +1,67 @@
+<template>
+  <div
+    class="flex flex-col w-full px-24px justify-center items-center align-middle"
+    :class="{'flex-grow': ['PASSED', 'OVERLIMIT'].includes(status)}"
+  >
+    <DebugCancelledAlert
+      v-if="status === 'CANCELLED'"
+      :total-specs="totalSpecs"
+      :total-skipped-specs="totalSkippedSpecs"
+      :canceled-at="canceledAt"
+      :canceled-by-full-name="canceledByFullName"
+      :canceled-by-email="canceledByEmail"
+    />
+    <DebugPassed v-else-if="status === 'PASSED'" />
+    <DebugErrored
+      v-else-if="status === 'ERRORED'"
+      :errors="errors"
+      :total-specs="totalSpecs"
+      :total-skipped-specs="totalSkippedSpecs"
+    />
+    <DebugNoTests v-else-if="status === 'NOTESTS'" />
+    <DebugTimedout
+      v-else-if="status === 'TIMEDOUT'"
+      :total-specs="totalSpecs"
+      :total-skipped-specs="totalSkippedSpecs"
+      :ci="ci"
+    />
+    <DebugOverLimit
+      v-else-if="status === 'OVERLIMIT' || isHiddenByUsageLimits"
+      :over-limit-action-type="overLimitActionType"
+      :over-limit-action-url="overLimitActionUrl"
+    />
+  </div>
+</template>
+<script lang="ts" setup>
+import { computed } from '@vue/reactivity'
+import type { CloudRunStatus, OverLimitActionTypeEnum } from '../generated/graphql'
+import DebugCancelledAlert from './DebugCancelledAlert.vue'
+import DebugPassed from './DebugPassed.vue'
+import DebugErrored from './DebugErrored.vue'
+import DebugNoTests from './DebugNoTests.vue'
+import DebugTimedout from './DebugTimedout.vue'
+import DebugOverLimit from './DebugOverLimit.vue'
+import type { CloudCiBuildInfo } from '@packages/data-context/src/gen/graphcache-config.gen'
+
+const props = defineProps<{
+  status: CloudRunStatus
+  canceledAt: string | null
+  canceledByFullName?: string | null
```

Comment:
If anyone has tips on how to deal with this, I'd be happy to refactor. I guess when something is nullable in GQL, it can be either `TheType | undefined | null`? This leads to weird stuff with TypeScript where it's not enough to just make the prop optional, you also have to specify that it could be `null`. It's ugly

Reply:
I spiked into fixing this but it's going to be a chunk of work. There's a bunch of options we can configure: https://the-guild.dev/graphql/codegen/plugins/typescript/typescript-operations

The ones we need to figure out and standardize:

- `maybeValue` (default is `T | null`, maybe we want `undefined`?)
- we should do `immutableTypes: false`. This will stop generating `readonly` which is a bit annoying, even if it's more technically correct

An alternative in the short term might be  `type Maybe<T> = T | undefined | null`. :man_shrugging: